ppa/archive/management/commands/update_hathi_pagecounts.py

```diff
@@ -0,0 +1,87 @@
+from collections import defaultdict
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.contrib.admin.models import CHANGE, LogEntry
+from django.contrib.auth.models import User
+from django.contrib.contenttypes.models import ContentType
+from pairtree import storage_exceptions
+from parasolr.django.signals import IndexableSignalHandler
+
+from ppa.archive.models import DigitizedWork
+
+
+class Command(BaseCommand):
+    """Update database page counts for non-excerpted HathiTrust digitized items.
+    By default, runs on all non-excerpted, public HathiTrust items.
+    """
+
+    help = __doc__
+
+    #: normal verbosity level
+    v_normal = 1
+    verbosity = v_normal
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "source_ids", nargs="*", help="List of specific items to update (optional)"
+        )
+
+    def handle(self, *args, **kwargs):
+        verbosity = kwargs.get("verbosity", self.v_normal)
+        source_ids = kwargs.get("source_ids", [])
+        # page count does not affect solr indexing, so disconnect signal handler
+        IndexableSignalHandler.disconnect()
+
+        script_user = User.objects.get(username=settings.SCRIPT_USERNAME)
+        digwork_contentype = ContentType.objects.get_for_model(DigitizedWork)
+
+        # find all non-excerpted, non-suppressed hathi volumes
+        hathi_vols = DigitizedWork.objects.filter(
+            source=DigitizedWork.HATHI,
+            item_type=DigitizedWork.FULL,
+            status=DigitizedWork.PUBLIC,
+        )
+        # if source ids are specified, limit to those records only
+        if source_ids:
+            hathi_vols = hathi_vols.filter(source_id__in=source_ids)
+
+        stats = defaultdict(int)
```

Comment:
For clarity, might it be worth fully declaring the desired dict structure upfront?
```suggestion
        stats = {"update":0, "unchanged":0}
```
This would also remove the use of `defaultdict`